backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/ReleaseService.java

```diff
@@ -19,22 +31,149 @@ public class ReleaseService implements ReleaseAnnouncementUseCase {
+  private final CreateNewVersionOfNormService createNewVersionOfNormService;
+  private final DeleteQueuedNormsPort deleteQueuedNormsPort;
+  private final DeleteUnpublishedNormPort deleteUnpublishedNormPort;
 
   public ReleaseService(
     LoadAnnouncementByNormEliUseCase loadAnnouncementByNormEliUseCase,
-    UpdateAnnouncementPort updateAnnouncementPort
+    UpdateAnnouncementPort updateAnnouncementPort,
+    UpdateOrSaveNormPort updateOrSaveNormPort,
+    NormService normService,
+    TimeMachineService timeMachineService,
+    CreateNewVersionOfNormService createNewVersionOfNormService,
+    DeleteQueuedNormsPort deleteQueuedNormsPort,
+    DeleteUnpublishedNormPort deleteUnpublishedNormPort
   ) {
     this.loadAnnouncementByNormEliUseCase = loadAnnouncementByNormEliUseCase;
     this.updateAnnouncementPort = updateAnnouncementPort;
+    this.updateOrSaveNormPort = updateOrSaveNormPort;
+    this.normService = normService;
+    this.timeMachineService = timeMachineService;
+    this.createNewVersionOfNormService = createNewVersionOfNormService;
+    this.deleteQueuedNormsPort = deleteQueuedNormsPort;
+    this.deleteUnpublishedNormPort = deleteUnpublishedNormPort;
   }
 
+  /**
+   * Queue all norms that are affected by the given announcement for publishing.
+   * This includes generating a new manifestation of the amending norm, and new expressions for each time boundary of
+   * every target norm. It also creates a new manifestation for the current expression of the target norm so it contains
+   * the information about the future changes and the correct "nachfolgende-version-id".
+   * <p></p>
+   * Previous releases of the same announcement of the current date are replaced by this release. Also, new
+   * unpublished manifestations are created for the amending norm and the current expression of the target norms to
+   * enable further work on the announcement.
+   * <p></p>
+   * NOTE: This is currently always creating new expressions for every time boundary (so what is necessary for the
+   * prätext-abgabe). When only metadata has changed to the previous abgabe no new expressions should be generated. This
+   * is not handled by this method yet.
+   *
+   * @param query The query specifying the {@link Announcement} to be loaded.
+   * @return The information about the announcement published.
+   */
   @Override
+  @Transactional
   public Announcement releaseAnnouncement(ReleaseAnnouncementUseCase.Query query) {
     var announcement = loadAnnouncementByNormEliUseCase.loadAnnouncementByNormEli(
       new LoadAnnouncementByNormEliUseCase.Query(query.eli())
     );
-    announcement.setReleasedByDocumentalistAt(Instant.now());
 
+    // find all norms to publish
+    Set<Norm> normsToPublish = new HashSet<>();
+    normsToPublish.add(normService.loadNorm(new LoadNormUseCase.Query(announcement.getEli())));
+
+    for (Norm norm : normsToPublish) {
+      var analysis = norm.getMeta().getAnalysis();
+      if (analysis.isEmpty()) {
```

Comment:
This should not happen, currently with delta approach, right? Should we even be allowing publishing an amending law if the analysis is empty? Or throw an error?

Reply:
its written in a general way so it can also handle publishing norms that are not amending norms. But yes in theory this should always exist for the amending norm